lib/transports.js

```diff
@@ -89,10 +122,47 @@ function HTTPSTransport(options) {
   this.options = options || {};
   this.agent = httpsAgent;
 }
+
+function HTTPProxyTransport(options) {
+  this.defaultPort = 80;
+  this.transport = http;
+  this.options = options || {};
+  this.agent = httpAgent;
+  this.options.host = options.proxyHost;
```

Comment:
How about just using `host` `port` etc. instead? If we pass `options` object to this constructor, it's by definition mentioned to be used by `proxy`, therefore using a prefix `proxyHost` seems redundant.

Reply:
Or it may be useful to distinguish `http` from `httpProxy` options. It's up to you really.